core/storage/user/gae_models.py

```diff
@@ -97,6 +97,85 @@ def get_by_normalized_username(cls, normalized_username):
             cls.normalized_username == normalized_username).get()
 
 
+class ActivitiesCompletedByLearnerModel(base_models.BaseModel):
+    """Keeps track of all the explorations and collections completed by the
+    learner.
+
+    Instances of this class are keyed by the user id.
+    """
+    # IDs of all the explorations completed by the user.
+    completed_exploration_ids = ndb.StringProperty(repeated=True, indexed=True)
+    # IDs of all the collections completed by the user.
+    completed_collection_ids = ndb.StringProperty(repeated=True, indexed=True)
+
+
+class IncompleteActivitiesModel(base_models.BaseModel):
+    """Keeps track of all the activities currently being completed by the
+    learner.
+
+    Instances of this class are keyed by the user id.
+    """
+    # The ids of the explorations partially completed by the user.
+    incomplete_exploration_ids = ndb.StringProperty(repeated=True, indexed=True)
+    # The ids of the collections partially completed by the user.
+    incomplete_collection_ids = ndb.StringProperty(repeated=True, indexed=True)
+
+
+class ExpUserLastPlaythroughModel(base_models.BaseModel):
+    """Stores the "last playthrough" information of the exploration since the
+    user left it partially completed.
```

Comment:
Let's be more explicit here and say 'Stores the "last playthrough" information for partially-completed explorations.", since you delete this after an exploration is completed.